src/agent/configuration_parser/include/configuration_parser.hpp

```diff
@@ -100,6 +109,10 @@ namespace configuration
                 {
                     return ParseTimeUnit(current.as<std::string>());
                 }
+                else if constexpr (std::is_same_v<T, std::size_t>)
+                {
+                    return ParseSizeUnit(current.as<std::string>());
+                }
```

Comment:
I would prefer to create a type (even if it's just a wrapper to an int or size_t) that it is clearly about bytes. Size_t is used for many things, now the user of the configuration parser needs to figure out somehow that when they ask for a size_t whatever they are getting will be converted to bytes.

Either that or have the ParseSizeUnit as a separate function outside the configuration parser and the user manually converts it.

Reply:
Using MessageSize